packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/PdfViewerTextLayer.tsx

```diff
@@ -0,0 +1,179 @@
+import useAsyncFunctionCall from './useAsyncFunctionCall';
+
+interface Props {
+  className?: string;
+
+  /**
+   * PDF page from pdfjs
+   */
+  loadedPage: PDFPageProxy | null | undefined;
+
+  /**
+   * Zoom factor, where `1` is equal to 100%
+   */
+  scale: number;
+
+  /**
+   * Callback for text layer info
+   */
+  setRenderedText?: (info: PdfRenderedText | null) => any;
+}
+
+export type PdfRenderedText = {
+  /**
+   * PDF text content
+   */
+  textContent: TextContent & {
+    styles: { [styleName: string]: CSSStyleDeclaration };
+  };
+
+  /**
+   * Text span DOM elements rendered on the text layer
+   */
+  textDivs: HTMLElement[];
+
+  /**
+   * Pdf page viewport used to render text items
+   */
+  viewport: PDFPageViewport;
+
+  /**
+   * Page number, starting at 1
+   */
+  page: number;
+};
+
+const PdfViewerTextLayer: FC<Props> = ({
+  className,
+  loadedPage,
+  scale = 1,
+  setRenderedText = () => {}
+}) => {
+  const textLayerRef = useRef<HTMLDivElement>(null);
+  const textLayerDiv = textLayerRef.current;
+
+  // load text content from the page
+  const loadedText = useAsyncFunctionCall(
+    useCallback(async () => {
+      if (loadedPage) {
+        const viewport = loadedPage.getViewport({ scale });
+        const textContent = await loadedPage.getTextContent();
+        return { textContent, viewport, page: loadedPage.pageNumber, scale };
+      }
+      return null;
+    }, [loadedPage, scale])
+  );
+
+  // render text content
+  const renderedText = useAsyncFunctionCall(
+    useCallback(
+      async (signal: AbortSignal) => {
+        if (textLayerDiv && loadedText) {
+          const { textContent, viewport, scale, page } = loadedText;
+
+          const builder = new TextLayerBuilder({
+            textLayerDiv,
+            viewport,
+            eventBus: new EventBus(),
+            pageIndex: page - 1
+          });
+          signal.addEventListener('abort', () => builder.cancel());
+
+          await _renderTextLayer(builder, textContent, textLayerDiv, scale);
+          return { textContent, viewport, page, textDivs: builder.textDivs };
+        }
+        return undefined;
+      },
+      [loadedText, textLayerDiv]
+    )
+  );
+
+  useEffect(() => {
+    if (renderedText !== undefined) {
+      setRenderedText(renderedText);
+    }
+  }, [renderedText, setRenderedText]);
+
+  const rootClassName = cx(className, `textLayer`);
+  return (
+    <div
+      className={rootClassName}
+      ref={textLayerRef}
+      style={{
+        width: `${loadedText?.viewport?.width ?? 0}px`,
+        height: `${loadedText?.viewport?.height ?? 0}px`
+      }}
+    />
+  );
+};
+
+/**
+ * Render text into DOM using the text layer builder
+ */
+async function _renderTextLayer(
+  builder: TextLayerBuilder,
+  textContent: TextContent,
+  textLayerDiv: HTMLDivElement,
+  scale: number
+) {
+  builder.setTextContent(textContent);
+
+  // render
+  textLayerDiv.innerHTML = '';
```

Comment:
Is this to clear out the contents of `textLayerDiv`?

Reply:
Yes it is. It's to clear text spans under `textLayerDiv` rendered by previous text layer rendering.